sympy/integrals/integrals.py

```diff
@@ -1391,6 +1391,42 @@ def integrate(*args, **kwargs):
     else:
         return integral
 
+def cauchypv(func, sym):
+    """cauchypv(function, symbol)
+
+    Compute the Cauchy Principle Value of a real function.
+
+    Cauchy Principle Value is  defined as:
+
+          +oo               +r
+         /                  / 
+         |                  |
+         |                  |  
+    p.v. | f(x)dx  = lim    | f(x)dx
+         |          r -> oo |
+         |                  | 
+         /                  /
+          -oo               -r
+    Examples
+    ========
+    >>> from sympy import *
+    >>> x = symbols('x', real = True)
+    >>> f = x
+    >>> cauchypv(f,x)
+    >>> 0
+    >>> g = x + 1
+    >>> cauchypv(g,x)
+    >>> nan
+    
+    """
+    r = symbols('r')
+    if sym.is_real and func.is_real:
```

Comment:
Instead of requiring real assumptions, you should check for `is_real == False` and only fail there (it should still work if `is_real == None`).